overlord/devicestate/handlers_install.go

```diff
@@ -949,42 +950,68 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	gi, err := gadget.ReadInfoAndValidate(mntPtForType[snap.TypeGadget], sys.Model, nil)
+	if err != nil {
+		return err
+	}
+
```

Comment:
should the TODO:

// TODO validation of onVolumes versus gadget.yaml, considering also partial

get closer to the code here now?

or is it actually done because MatchDisksToGadgetVolumes below now uses EnsureVolumeCompatibility ?

Reply:
I've moved the TODO down, a compatibility check between the gadget and what we take from the installer needs still to happen.